woocommerce/woocommerce-ios
# SPE: Present new action sheet for product creation

Part of: #8836


## Description


This adds a new action sheet for product creation when the Simplified Product Editing feature flag is enabled. It combines template and manual product types in one sheet.

New store owners (no products) see these options:

* Physical (simple, non-virtual) - created with a template
* Virtual (simple, virtual) - created with a template
* External - created with a template
* Blank (simple, non-virtual) - **no** template

Existing store owners with products will also see:

* Grouped - created with a template

A couple caveats:

1. The designs include a separator between `External product` and `Blank`. This requires some more significant changes to the bottom sheet so I will address that in a later PR.
2. Selecting "Grouped product" imports several other products, due to the template product data (see https://github.com/woocommerce/woocommerce-ios/issues/8836#issuecomment-1440048492). I'll address that in a separate PR.

### Changes

* A new image asset is added for the blank product icon.
* The blank product type is added to `BottomSheetProductType`, and some of the product type strings are changed under the feature flag.
* In `AddProductCoordinator` we conditionally show the new action sheet when the feature flag is enabled, with the behavior described above.

## Testing instructions


1. Build and run the app (with the feature flag enabled).
2. Select a store with no products.
3. Go to the Products tab and tap "+" to create a new product.
4. Confirm the new action sheet appears with the expected product types (excluding grouped products).
5. Select a product type and confirm the product details screen appears as expected.
6. Save or publish a product.
7. Create another new product and confirm the action sheet now shows the grouped product type.

## Screenshots


/|Before|After
-|-|-
No products|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 36 04](https://user-images.githubusercontent.com/8658164/220673060-9347fc92-36e4-49a4-a8c5-5e1ff8e5ad79.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 31 57](https://user-images.githubusercontent.com/8658164/220671838-bcdc572e-a046-411e-a28c-690d9ae1ebfb.png)
With products|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 35 41](https://user-images.githubusercontent.com/8658164/220673049-181f2757-b91d-4602-a633-26e8f8425fb3.png)|![Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 32 09](https://user-images.githubusercontent.com/8658164/220671858-69afbbd6-ca90-43f2-9dd5-3855583ea6b2.png)

https://user-images.githubusercontent.com/8658164/220668077-af82b0fa-18e4-4ccf-b18a-94140d06d3fb.mp4



---
